### ISSUE TITLE
[package] libxml2 does not install libxml.m4

### ISSUE THREAD

**Issue by madebr** [conan-io/conan-center-index#6769]
libxml2 provides a m4 script for other project making use of autoconf based build scripts.
The current libxml2 recipe does not install this m4 script + adds its path to `AUTOMAKE_CONAN_INCLUDES`.

https://gitlab.gnome.org/GNOME/libxml2/-/blob/master/libxml.m4

This (potential) problem was found in https://github.com/conan-io/conan-center-index/pull/6768#discussion_r684841712

**Comment by SSE4** [comment]
just in case, does this m4 file contain hard-coded paths?
this is a typical situation some projects install .m4 files. do we need some special handling for them in conan (like, `m4_paths` maybe?)? or, perhaps, `res_paths` is enough with custom env var?

**Comment by madebr** [comment]
No, these M4 files are meant to be embedded in configure scripts when running `autoreconf`.
Embedded paths would be counterintuitive to the auto detection purpose of autoconf and descendants.
